src/Features/Core/Portable/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs

```diff
@@ -0,0 +1,239 @@
+                    {
+                        messageFormat = FeaturesResources.Remove_unused_parameter_0_its_initial_value_is_never_used;
+                    }
+                    else
+                    {
+                        messageFormat = s_unusedParameterRule.MessageFormat;
+                    }
+
+                    return new DiagnosticHelper.LocalizableStringWithArguments(messageFormat, parameter.Name);
+                }
+            }
+
+            private static IEnumerable<INamedTypeSymbol> GetAttributesForMethodsToIgnore(Compilation compilation)
+            {
+                // Ignore conditional methods (One conditional will often call another conditional method as its only use of a parameter)
+                var conditionalAttribte = compilation.ConditionalAttribute();
+                if (conditionalAttribte != null)
+                {
+                    yield return conditionalAttribte;
+                }
+
+                // Ignore methods with special serialization attributes (All serialization methods need to take 'StreamingContext')
+                var onDeserializingAttribute = compilation.OnDeserializingAttribute();
+                if (onDeserializingAttribute != null)
+                {
+                    yield return onDeserializingAttribute;
+                }
+
+                var onDeserializedAttribute = compilation.OnDeserializedAttribute();
+                if (onDeserializedAttribute != null)
+                {
+                    yield return onDeserializedAttribute;
+                }
+
+                var onSerializingAttribute = compilation.OnSerializingAttribute();
+                if (onSerializingAttribute != null)
+                {
+                    yield return onSerializingAttribute;
+                }
+
+                var onSerializedAttribute = compilation.OnSerializedAttribute();
+                if (onSerializedAttribute != null)
+                {
+                    yield return onSerializedAttribute;
+                }
+
+                // Don't flag obsolete methods.
+                var obsoleteAttribute = compilation.ObsoleteAttribute();
+                if (obsoleteAttribute != null)
+                {
+                    yield return obsoleteAttribute;
+                }
+            }
+
+            private bool IsUnusedParameterCandidate(IParameterSymbol parameter)
+            {
+                // Ignore certain special parameters/methods.
+                // Note that "method.ExplicitOrImplicitInterfaceImplementations" check below is not a complete check,
+                // as identifying this correctly requires analyzing referencing projects, which is not
+                // supported for analyzers. We believe this is still a good enough check for most cases so 
+                // we don't have to bail out on reporting unused parameters for all public methods.
+
+                if (parameter.IsImplicitlyDeclared ||
```

Comment:
nit: empty line looks inconsistent here

Reply:
That's personally fine with me.  I like comments that abut things they directly affect, and are separated when tehy're documenting a bunch of stuff.